surfactant/filetypeid/id_magic.py

```diff
@@ -28,7 +28,22 @@ def identify_file_type(filepath: str) -> Optional[str]:
             if magic_bytes[:4] == b"\x7fELF":
                 return "ELF"
             if magic_bytes[:2] == b"MZ":
+                # Check to see if this is actually a PE file or a DOS file
+                coff_addr = (
+                    int.from_bytes(magic_bytes[0x3C:0x40], byteorder="little", signed=False)
+                    & 0xFFFF
+                )
+                if magic_bytes[coff_addr : coff_addr + 2] != b"PE":
+                    return "DOS"
```

Comment:
```suggestion
                if magic_bytes[coff_addr : coff_addr + 4] != b"PE\x00\x00":
                    return "DOS"
```

It looks like the expected magic for a PE file is `PE\x00\x00` -- it would be interesting to know if Windows will accept a PE file without those two trailing null-bytes as part of the magic bytes or not.

The other potential corner case would be if the coff_addr it gets from the file points to a location beyond the initial 265 byte read; it seems like the expected coff_addr is basically a fixed value, so this would be another unlikely case that would only appear in an intentionally odd or malformed PE file.